osier/equations.py

```diff
@@ -0,0 +1,160 @@
+from abc import ABC, abstractmethod
+import numpy as np
+
+class OsierEquation(ABC):
+
+    def __init__(self) -> None:
+        pass        
+
+    @abstractmethod
+    def _do(self, technology_list, X):
+        pass
+
+
+def get_tech_names(technology_list):
+    """
+    Parameters
+    ----------
+    technology_list : list of :class:`osier.Technology` objects
+        The list of technologies.
+    dispatch_results : :class:`pandas.DataFrame`
+        The results dispatch results for each technology in 
+        :attr:`technology_list` from a :class:`osier.DispatchModel` run.
+    """
+
+    tech_names = [t.technology_name for t in technology_list]
+
+    return tech_names
+
+
+def get_dispatchable_techs(technology_list):
+    """
+    Parameters
+    ----------
+    technology_list : list of :class:`osier.Technology` objects
+        The list of technologies.
+    dispatch_results : :class:`pandas.DataFrame`
+        The results dispatch results for each technology in 
+        :attr:`technology_list` from a :class:`osier.DispatchModel` run.
+    """
+
+    dispatchable_techs = [t for t in technology_list if t.dispatchable]
+
+    return dispatchable_techs
+
+
+def get_nondispatchable_techs(technology_list):
+    """
+    Parameters
+    ----------
+    technology_list : list of :class:`osier.Technology` objects
+        The list of technologies.
+    dispatch_results : :class:`pandas.DataFrame`
+        The results dispatch results for each technology in 
+        :attr:`technology_list` from a :class:`osier.DispatchModel` run.
+    """
+
+    dispatchable_techs = [t for t in technology_list if not t.dispatchable]
+
+    return dispatchable_techs    
+
+
+def get_dispatchable_names(technology_list):
+    """
+    Parameters
+    ----------
+    technology_list : list of :class:`osier.Technology` objects
+        The list of technologies.
+    dispatch_results : :class:`pandas.DataFrame`
+        The results dispatch results for each technology in 
+        :attr:`technology_list` from a :class:`osier.DispatchModel` run.
+    """
+
+    dispatchable_techs = [t.technology_name for t in technology_list if t.dispatchable]
+
+    return dispatchable_techs
```

Comment:
It looks like you're omitting the `Return` docstring on these `get_*` functions. Does PEP8 allow this? If so, then the docstrings are fine as is as the functions themselves are very readable. If not, you should specify the type of variable returned by the `get_*` functions in their docstrings.

Reply:
I'll specify the return variables in the doc string.